clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -11486,6 +11486,13 @@ void LinkerWrapper::ConstructJob(Compilation &C, const JobAction &JA,
     if (Args.hasArg(options::OPT_fsycl_embed_ir))
       CmdArgs.push_back(Args.MakeArgString("-sycl-embed-ir"));
 
+    if (Args.hasArg(options::OPT_fsycl_allow_device_image_dependencies))
+      CmdArgs.push_back(
+          Args.MakeArgString("-sycl-allow-device-image-dependencies"));
+    if (Args.hasArg(options::OPT_fno_sycl_allow_device_image_dependencies))
+      CmdArgs.push_back(
+          Args.MakeArgString("-no-sycl-allow-device-image-dependencies"));
+
```

Comment:
```suggestion
    if (Args.hasFlag(options::OPT_fsycl_allow_device_image_dependencies, options::OPT_fno_sycl_allow_device_image_dependencies, false))
      CmdArgs.push_back(
          Args.MakeArgString("-sycl-allow-device-image-dependencies"));
    if (Args.hasArg(options::OPT_fno_sycl_allow_device_image_dependencies))

```
Single `hasFlag` should do the trick, then there won't be the case of both flags being passed.

Reply:
Followed your suggestion and added tests to `clang/test/Driver/sycl-offload-new-driver.c`. Is that good enough or do you prefer a new separate test?